Source/Core/Core/HW/GCMemcard/GCMemcard.cpp

```diff
@@ -1456,17 +1456,25 @@ Header::Header(int slot, u16 size_mbits, bool shift_jis)
   calc_checksumsBE((u16*)this, 0xFE, &m_checksum, &m_checksum_inv);
 }
 
+static u32 InterpretAsU32LE(std::byte b0, std::byte b1, std::byte b2, std::byte b3)
```

Comment:
This should probably go into some utility class, right?

Reply:
You could probably use `swap32(const u8* data)` for this.

Reply:
Yeah I'm not sure about that. If I'm reading the standard right that's still undefined behavior if `u8` does not map to `char`, `unsigned char`, or `std::byte`. Granted, that is pretty unlikely to happen on anything but esoteric embedded hardware that can't possibly run Dolphin anyway, but y'know, if we're already rewriting it to do it right...

Reply:
So I tried the checksum calculation with `u8` instead and yeah, that does come out a lot more readable. I dunno. I'm fine with either.

Reply:
> Yeah I'm not sure about that. If I'm reading the standard right that's still undefined behavior if u8 does not map to `char`, `unsigned char`, or `std::byte`

Dolphin would not run at all on platforms where this isn't the case (just as we wouldn't run on platforms where `float` and `double` aren't 32-bit and 64-bit respectively)

Reply:
Okay, removed the whole thing and just using u8.